test/statistic_test.cc

```diff
@@ -289,25 +315,39 @@ TEST(StatisticTest, HdrStatisticPercentilesProto) {
   EXPECT_EQ(json, golden_json);
 }
 
+TEST(StatisticTest, CircllhistStatisticPercentilesProto) {
+  nighthawk::client::Statistic parsed_json_proto;
+  CircllhistStatistic statistic;
+
+  for (int i = 1; i <= 10; i++) {
+    statistic.addValue(i);
+  }
+
+  Envoy::MessageUtil util;
+  util.loadFromJson(Envoy::Filesystem::fileSystemForTest().fileReadToEnd(
+                        TestEnvironment::runfilesPath("test/test_data/circllhist_proto_json.gold")),
+                    parsed_json_proto, Envoy::ProtobufMessage::getStrictValidationVisitor());
+  // Instead of comparing proto's, we perform a string-based comparison, because that emits a
```

Comment:
A more idiomatic approach would be to compare the proto and output a textual diff only if the comparison fails. That way we will be comparing via the binary representations that are the public API. The string are only for humans to read.

Note that EXPECT_EQ takes in an optional stream which can contain a custom error message - this is where we can inject the readable diff:

https://github.com/google/googletest/blob/master/googletest/docs/primer.md#assertions

Reply:
Changed to compare the proto. PTAL.